backend/endpoints/rooms.py

```diff
@@ -50,48 +33,60 @@ async def create_room(
         member_id: A unique identifier of the member creating the room
 
     Returns:
-        A dict mapping the key room_id to the unique identifier (id) value of the room
-            {"room_id: "_id"}
-
-    Raises:
-        HTTP_424_FAILED_DEPENDENCY: "room creation unsuccessful" or
-            "unable to read database"
-        HTTP_406_NOT_ACCEPTABLE: raised when the maximium number of allowed users is exceeded
+        HTTP_200_OK (room already exist): {room_id}
+        HTTP_201_CREATED (new room created): {room_id}
+    Raises
+        HTTP_424_FAILED_DEPENDENCY: room creation unsuccessful
     """
     room_data = request.dict()
-    room_members = set(room_data["room_members"])
-    if len(room_members) > 9:
+    plugin_name = room_data.get("plugin_name")
+    room_name = room_data.get("room_name")
+    room_member_ids = list(room_data.get("room_members").keys())
+    rooms = await get_org_rooms(org_id=org_id, plugin=plugin_name)
+
+    if rooms is not None:
+        if plugin_name == Plugin.CHANNEL.value and room_name.casefold() in [
+            room["room_name"].casefold() for room in rooms
+        ]:
+            raise HTTPException(
+                status_code=status.HTTP_200_OK, detail={"room_name": room_name}
+            )
+
+        if plugin_name == Plugin.DM.value:
+            for room in rooms:
+                if set(room["room_members"].keys()) == set(room_member_ids):
+                    raise HTTPException(
+                        status_code=status.HTTP_200_OK,
+                        detail={"room_id": room["room_id"]},
+                    )
+
+        response = await DB.write("rooms", data=room_data)
+        if response and response.get("status") == 200:
+            room_id = {"room_id": response.get("data").get("object_id")}
+            centrifugo_data = await sidebar.format(
+                org_id,
+                member_id,
+                plugin=plugin_name,
+            )  # getting the response data
+
+            background_tasks.add_task(
```

Comment:
Can we extract out this part? It could be used in other places too...

Reply:
Well fastapi handles the background task specially which is why it has a parameter on the function.
Separating to a function of it's own, might make seperate the fastapi handling of the task.

I would look for better ways of handling it

Reply:
Ok. We'll look for better ways then.